Python/thread_pthread.h

```diff
@@ -39,33 +41,35 @@
  * The default stack sizes below are the empirically determined minimal stack
  * sizes where a simple recursive function doesn't cause a hard crash.
  */
-#if defined(__APPLE__) && defined(THREAD_STACK_SIZE) && THREAD_STACK_SIZE == 0
+#if defined(__APPLE__) && THREAD_STACK_SIZE == -1
 #undef  THREAD_STACK_SIZE
 /* Note: This matches the value of -Wl,-stack_size in configure.ac */
 #define THREAD_STACK_SIZE       0x1000000
 #endif
-#if defined(__FreeBSD__) && defined(THREAD_STACK_SIZE) && THREAD_STACK_SIZE == 0
+#if defined(__FreeBSD__) && THREAD_STACK_SIZE == -1
 #undef  THREAD_STACK_SIZE
 #define THREAD_STACK_SIZE       0x400000
 #endif
-#if defined(_AIX) && defined(THREAD_STACK_SIZE) && THREAD_STACK_SIZE == 0
-#undef  THREAD_STACK_SIZE
-#define THREAD_STACK_SIZE       0x200000
-#endif
 /* bpo-38852: test_threading.test_recursion_limit() checks that 1000 recursive
    Python calls (default recursion limit) doesn't crash, but raise a regular
    RecursionError exception. In debug mode, Python function calls allocates
    more memory on the stack, so use a stack of 8 MiB. */
-#if defined(__ANDROID__) && defined(THREAD_STACK_SIZE) && THREAD_STACK_SIZE == 0
+#if defined(__ANDROID__) && THREAD_STACK_SIZE == -1
 #   ifdef Py_DEBUG
 #   undef  THREAD_STACK_SIZE
 #   define THREAD_STACK_SIZE    0x800000
 #   endif
 #endif
-#if defined(__VXWORKS__) && defined(THREAD_STACK_SIZE) && THREAD_STACK_SIZE == 0
+#if defined(__VXWORKS__) && THREAD_STACK_SIZE == -1
 #undef  THREAD_STACK_SIZE
 #define THREAD_STACK_SIZE       0x100000
 #endif
+#if THREAD_STACK_SIZE == -1
+#undef  THREAD_STACK_SIZE
+/* This is big enough for most platforms without being wasteful. */
```

Comment:
do we have a unittest in the stdlib that exercises the default recursion limit to **on a thread** to ensure the process doesn't crash?  _(it'll be imperfect as different internal call recursion codepaths have different C stack needs, but it still serves as a good failsafe; perhaps have the test raise recursionlimit to 1.5x its default value)_.

https://github.com/python/cpython/blob/v3.10.0/Lib/test/test_threading.py#L1234 test_threading.test_recursion_limit exists.  We should probably change that to raise the recursion limit a bit above the default in it's thread for some safety margin.